core/vm/evm.go

```diff
@@ -328,7 +329,11 @@ func (evm *EVM) Call(caller common.Address, addr common.Address, input []byte, g
 				evm.Config.Tracer.OnGasChange(gas, 0, tracing.GasChangeCallFailedExecution)
 			}
 
-			usedMultiGas.SaturatingIncrementInto(multigas.ResourceKindComputation, gas)
+			// Attribute all leftover gas to computation
+			if startingGas > usedMultiGas.SingleGas() {
+				leftOverGas := startingGas - usedMultiGas.SingleGas()
+				usedMultiGas.SaturatingIncrementInto(multigas.ResourceKindComputation, leftOverGas)
+			}
```

Comment:
I didn't get this change. The existing gas variable should be left over gas, isn't it?

Reply:
Initially, the test results did not match, so I left this form as more informative. I returned usedMultiGas.SaturatingIncrementInto(multigas.ResourceKindComputation, gas)` back.